ranges.go

```diff
@@ -0,0 +1,62 @@
+package lipgloss
+
+import (
+	"strings"
+
+	"github.com/charmbracelet/x/ansi"
+)
+
+// StyleRanges allows to, given a string, style a range of it differently.
+// The function will take into account existing styles.
+// See [StyleRanges] to style multipe ranges in the same string.
+func StyleRange(s string, start, end int, style Style) string {
```

Comment:
Instead, maybe we could do `StyleRanges(s, NewRange(start, end, style))` where `func StyleRanges(s string, ranges ...Range) string` 🤔

Reply:
yeah sounds good to me as well :)

Reply:
Yeah, great call @aymanbagabas. Part of me thinks it should be singular, though, just so users don't think it must take multiple values 🤔. But maybe I'm overthinking it.

Reply:
fwiw I'm more inclined into using plural, as it indicates one or more, whereas singular indicates only one...